fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/data/LoanSummaryData.java

```diff
@@ -144,6 +146,7 @@ public static LoanSummaryData withTransactionAmountsSummary(final LoanSummaryDat
             totalRepaymentTransactionReversed = computeTotalAmountForReversedTransactions(LoanTransactionType.REPAYMENT, loanTransactions);
             totalInterestPaymentWaiver = computeTotalAmountForNonReversedTransactions(LoanTransactionType.INTEREST_PAYMENT_WAIVER,
                     loanTransactions);
+            totalInterestRefund = computeTotalAmountForNonReversedTransactions(LoanTransactionType.INTEREST_REFUND, loanTransactions);
```

Comment:
I think we should move away from using the "all loan transactions" (there might be cases when we are not fetching the loan transactions at all) to have a query to fetch the proper amount instead. As the loan got more and more loan transactions it is becoming a performance pitfall.

Would you mind to rewrite this to fetch either filtered transactions (only Interest refund types) or calculate immediately the total amount of interest refund transactiosn (that are not reverted)?

Reply:
Done! Loan Summary updated as we commented